sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java

```diff
@@ -186,21 +195,26 @@ private ShareFileAsyncClient createShareFileAsyncClient(ResourceUploadDefinition
     }
 
     private Mono<Void> uploadToStorage(File source, ResourceUploadDefinition option) {
-        innerModel().properties().source().withRelativePath(option.relativePath());
-        try {
-            ShareFileAsyncClient shareFileAsyncClient = createShareFileAsyncClient(option);
-            return shareFileAsyncClient.create(source.length())
-                .flatMap(fileInfo -> shareFileAsyncClient.uploadFromFile(source.getAbsolutePath()))
-                .then(Mono.empty());
-        } catch (Exception e) {
-            return Mono.error(e);
+        UserSourceInfo userSourceInfo = innerModel().properties().source();
+        if (userSourceInfo instanceof UploadedUserSourceInfo) {
+            UploadedUserSourceInfo uploadedUserSourceInfo = (UploadedUserSourceInfo) userSourceInfo;
+            try {
+                uploadedUserSourceInfo.withRelativePath(option.relativePath());
+                ShareFileAsyncClient shareFileAsyncClient = createShareFileAsyncClient(option);
+                return shareFileAsyncClient.create(source.length())
+                    .flatMap(fileInfo -> shareFileAsyncClient.uploadFromFile(source.getAbsolutePath()))
+                    .then(Mono.empty());
+            } catch (Exception e) {
+                return Mono.error(e);
+            }
+        } else {
+            return Mono.empty();
```

Comment:
What's other UserSourceInfo do?

Reply:
It's mainly for newly added `BuildResultUserSourceInfo` which requires setting `relativePath` in a different way. I removed this branch in future PR.